velox/functions/sparksql/aggregates/AverageAggregate.cpp

```diff
@@ -74,6 +75,37 @@ class AverageAggregate
   }
 };
 
+TypePtr getAvgType(const TypePtr& rawInputType) {
+  auto [p, s] = getDecimalPrecisionScale(*rawInputType.get());
+  // In Spark,
+  // sum precision = input precision + 4
+  // sum scale = input scale + 4
```

Comment:
It is a bit confusing because the function name is `getAvgType` but the comments refer to `sum`. Is intermediate type for sum is returned, and would you please add comments for this function to clarify?

Reply:
This function is used to get the result type of avg, the comments should be avg precision = ... and avg scale = ..., consistent with https://github.com/apache/spark/blob/a8e35c407bc5340f83b35e5a2f0b0767c6baadb0/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala#L69. I will change the function name to getAvgResultType.